src/pages/nonprofit/issues/create.tsx

```diff
@@ -1,9 +1,225 @@
+import { Button } from "@chakra-ui/button";
+import { Heading, Stack, Text, VStack, Link } from "@chakra-ui/layout";
+import { Radio, RadioGroup } from "@chakra-ui/radio";
 import { Flex } from "@chakra-ui/react";
+import { Textarea } from "@chakra-ui/textarea";
+import { useEffect, useState } from "react";
+import { HiOutlineChevronLeft } from "react-icons/hi";
+import { nonprofitCreateIssue } from "src/actions/Issue";
+import { nonprofitGetProject } from "src/actions/Project";
+import { showError, showInfo } from "src/utils/notifications";
+import {
+  Chapter,
+  IssueStatus,
+  MaintenanceType,
+  NonprofitCreateIssue,
+  Project,
+  ProjectStage,
+  UploadedFile,
+} from "src/utils/types";
+import { uploadFile } from "src/utils/uploaded-files";
+import urls from "src/utils/urls";
 
 function NonprofitIssueCreationPage() {
+  const [title, setTitle] = useState("");
+  const [description, setDescription] = useState("");
+  const [issueType, setIssueType] = useState<MaintenanceType>();
+  const [project, setProject] = useState<Project>();
+  const [preUploadImages, setPreUploadImages] = useState<FileList>();
+  const [images, setImages] = useState<UploadedFile[]>([]);
+  const [chapterIssueTypes, setChapterIssueTypes] = useState<MaintenanceType[]>(
+    [MaintenanceType.BUG_FIXES, MaintenanceType.NEW_FEATURES]
+  );
+  const [projectId, setProjectId] = useState("617ea71c069ce109e2ae9f83");
+
+  useEffect(() => {
+    async function getProject() {
+      let project;
+      try {
+        project = await nonprofitGetProject(projectId, {
+          status: ProjectStage.MAINTENANCE,
+        });
+      } catch (e) {
+        showError((e as Error).message);
+      }
+      if (!project) {
+        showError("Project does not exist.");
+        return;
+      }
+
+      setProject(project);
+    }
+
+    getProject()
+      .then(() => {
+        if (project) {
+          /**
+           * project.chapter as Chapter does not currently work either but when it will change:
+           * setChapterIssueTypes([MaintenanceType.BUG_FIXES, MaintenanceType.NEW_FEATURES]);
+           * to
+           * setChapterIssueTypes(chapter.maintenanceTypes;);
+           */
+          const chapter = project.chapter as Chapter;
+          setChapterIssueTypes([
+            MaintenanceType.BUG_FIXES,
+            MaintenanceType.NEW_FEATURES,
+          ]);
+        }
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  }, []);
+  const uploadImages = async ({ imageList }: { imageList: FileList }) => {
+    if (!imageList){
+      return;
+    }
+
+    if (imageList.length > 4) {
+      showError("Cannot upload more than 4 images!");
+    }
+
+    for (let i = 0; i < imageList.length; i++) {
+      const result = await uploadFile(imageList[i], {
+        onProgress(percent) {
+          console.log(percent);
+        },
+      });
+      setImages([...images, result]);
```

Comment:
Correct me if I am wrong, but there should not be a need to set the images after uploading right?